app/lib/resources/preprocessor.rb

```diff
@@ -0,0 +1,65 @@
+module Resources
+  # Takes a bunch of unsafe, string-y data that came in over AMQP and parses it
+  # into fully formed
+  class PreProcessor < Mutations::Command
+    def self.from_amqp(delivery_info, body)
+      # ["bot", "device_3", "resources_v0", "destroy", "Sequence", "2", "123-456
```

Comment:
This comment seems unfinished?

Reply:
That was a "note to self" about what `delivery_info.routing_key.split(".")` looks like at runtime.

Here's a more in-depth explanation:

```ruby
      # -- Snip! --
      # Parse the AMQP rotuing key into an Array of strings.
      # A properly formatted routing_key will look like this after processing:
      # 
      # ["bot", "device_3", "resources_v0", "destroy", "Sequence", "2", "xyz"]
      segments = delivery_info.routing_key.split(".")
      _, device_name, _, action, resource, resource_id, uuid = segments
     # -- Snip! --
```